util/http.py

```diff
@@ -349,6 +355,7 @@ def debuggable_request(cls, http_method, url, make_request_with=None,
         """
         logging.info("Making debuggable %s request to %s: kwargs %r",
                      http_method, url, kwargs)
+        make_request_with = make_request_with or requests.request
```

Comment:
This is the problem that only showed up in real usage. When you don't mock `make_request_with`, no real value was being set.